### PR TITLE
backup: add etcdClientWithMaxRevision

### PR DESCRIPTION
**Pull request by fanminshi**
refactor get max member and its client into a method called etcdClientWithMaxRevision.

### CODE REVIEW

**Comment by fanminshi** [comment]
cc/ @hongchaodeng @hasbro17

File: pkg/backup/backup_manager.go [hunk 1]
```diff
@@ -59,39 +59,18 @@ func NewBackupManager(kubecli kubernetes.Interface, clusterName string, namespac
 // SaveSnap saves the latest snapshot if its revision is greater than the given lastSnapRev
 // and returns a BackupStatus containing saving backup metadata if SaveSnap succeeds.
 func (bm *BackupManager) SaveSnap(lastSnapRev int64) (*backupapi.BackupStatus, error) {
-	podList, err := bm.kubecli.Core().Pods(bm.namespace).List(k8sutil.ClusterListOpt(bm.clusterName))
+	etcdcli, rev, err := bm.maxMemberCli()
```
Comment (hongchaodeng) [inline pkg/backup/backup_manager.go]: etcdClientWithMaxRevision()

File: pkg/backup/backup_manager.go [hunk 2]
```diff
@@ -59,39 +59,18 @@ func NewBackupManager(kubecli kubernetes.Interface, clusterName string, namespac
...
-		}
-	}
-
-	if len(pods) == 0 {
-		return nil, errors.New("no running etcd pods found")
-	}
-	member, rev := getMemberWithMaxRev(pods, bm.etcdTLSConfig)
-	if member == nil {
-		return nil, errors.New("no reachable member")
+		return nil, fmt.Errorf("create etcd client failed: %v", err)
```
Comment (hongchaodeng) [inline pkg/backup/backup_manager.go]: create etcd client with max revision failed:

File: pkg/backup/backup_manager.go (hunk 1 above)
Comment (fanminshi) [reply pkg/backup/backup_manager.go]: done!

File: pkg/backup/backup_manager.go (hunk 2 above)
Comment (fanminshi) [reply pkg/backup/backup_manager.go]: done!

**Comment by hasbro17** [comment]
LGTM

**Comment by hongchaodeng** [comment]
lgtm

**Pull request merged**